frontend/src/utils/use_state_delayed.ts

```diff
@@ -0,0 +1,24 @@
+import { useState } from 'react';
+
+export function useStateDelayed(
+  initial: boolean,
+  config = { late: 25, prolong: 200 },
+): [{ now: boolean; prolonged: boolean; late: boolean }, (next: boolean) => void] {
```

Comment:
Can i have a short comment what this does please? The variable names don't really help me understand it